docs/src/index.md

```diff
@@ -2,6 +2,8 @@
 EditURL = "https://github.com/JuliaIO/Tar.jl/blob/master/docs/src/index.md"
 ```
 
+The `Tar` module provides a simple interface for handling tar archives.
+
 # Tar
 
 ```@docs
```

Comment:
I notice that none of these functions are currently exported.    In Julia 1.11, they should be declared public, e.g. something like:
```jl
@static if VERSION >= v"1.11"
    public create, extract, list, rewrite, tree_hash, Header
end
```
since this is apparently the documented API.

Reply:
I think this is out of scope of this PR. I'll reference it in https://github.com/JuliaLang/julia/issues/51335 instead to not forget.

Reply:
> I'm not sure if there is a way to avoid `eval(Meta.parse(...))` since the public keyword does not parse in earlier Julia versions?

The only good way to avoid that that I know of is to use Compat.jl's `@compat public foo, bar`, though this is a fine use of `eval(Meta.parse(....))` imo.

Reply:
Or make a small macro here `macro public(v); @assert isexpr(v, :tuple); VERSION ? esc(Expr(:public, v.args...)); nothing; end` so that it just needs to add an `@` to make it syntactically valid here

Reply:
That macro (with some bug-fixes) is here https://github.com/JuliaLang/Compat.jl/pull/805/files/0c90fae7a869a5f526a4cb13620ea077fc30456e, though it was renamed to `@compat public`.

Reply:
Can we get this PR here merged? The discussion about declaring public symbols can be moved to #174 and/or #175. There I already implemented two different approaches